osmdroid-android/src/main/java/org/osmdroid/views/MapController.java

```diff
@@ -471,8 +473,15 @@ protected void onAnimationStart() {
     }
 
     protected void onAnimationEnd() {
+        final double newZoom = mMapView.mTargetZoomLevel.get();
+        final Projection newProjection = mMapView.getProjection().getOffspring(
+                newZoom, mMapView.getProjection().getScreenRect());
+        final IGeoPoint newCenter = newProjection.fromPixels(
```

Comment:
possibly related to the open issue. not saying there's something wrong with this, just leaving myself a note to look at this in the near future

Reply:
This code is 95% correct. The shift @Maradox and I experienced is only something like a 5 pixel shift, just after the end of the zoom animation (where I need to say: "now that the animation is done, let's center on that GeoPoint").
I need to understand more about the animation in order to compute (or retrieve) the animation's last map center.

Reply:
Just committed a fix:
* in fact the zoom-in animation refreshes the `MapView`'s `Projection` many times, with increasing values of scales from 1f to 2f - the question being how to get the correct map center after the animation
* in the constructor of `Projection`, I now compute a new member (`mCurrentCenter`) - the screen center as a `GeoPoint`
* after the zoom-in animation, I reuse the previous `Projection`'s `mCurrentCenter` for my `MapView` refresh

@Maradox Do you find this version convincing?